Add support for Dotenv v3, v4, and v5

mcaskill opened the pull request:
## Checklist:
- [x] I've read the [Contributing page](https://github.com/junaidbhura/composer-wp-pro-plugins/blob/master/CONTRIBUTING.md).
- [ ] I've created an issue and referenced it here.
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

## Description

Adds support for Dotenv [v5](https://github.com/vlucas/phpdotenv/releases/tag/v5.0.0) while also keeping support for v4 and restoring support for v3.

This allows older projects that might be stuck on an older version of Dotenv to continue to benefit from updates to this Composer plugin.

## How has this been tested?

I'm testing this on a client project that uses Advanced Custom Fields Pro, Gravity Forms, and Polylang Pro.

## Types of changes

#### Added

- Method `Installer::loadDotenv()` to resolve and instantiate the currently available version of the `Dotenv` class.
- Class `AbstractPlugin` as basis for all supported WP plugins. The class provides a `get()` method which serves as a wrapper of `getenv()` that will throw an exception if the requested environment variable is missing.
- Class `MissingEnvException` in case an environment variable is missing.

#### Changed

- WP plugin classes to extend new `AbstractPlugin` and use its `get()` method instead of directly using `getenv()` function.

junaidbhura commented:
Thanks @mcaskill ! As always, much appreciated!

Let me take a deeper look and get back to you!

junaidbhura commented on Installer.php:
```diff
@@ -37,12 +37,44 @@ public function activate( Composer $composer, IOInterface $io ) {
+	 *
+	 * @param string $path
+	 *
+	 * @return Dotenv|null
```
@mcaskill Do we need to return anything in this function?

junaidbhura commented:
Hey @mcaskill can we limit this PR to just the changes in `Installer.php` and `composer.json`?

I'm looking to do a big refactor of this plugin in the coming months, which will implement an Abstract class, etc.

mcaskill committed:
Add support for vlucas/phpdotenv v3 to v5

Added:
- Support for Dotenv v3, v4, or v5.
- Method `Installer::loadDotenv()` to resolve the current version and instantiate the available `Dotenv` class.

mcaskill commented:
Sorry for the delay and thanks for the feedback.

junaidbhura commented:
Thanks @mcaskill really appreciate it as always 👍
